bionty/_entity.py

```
@@ -290,7 +309,7 @@ def _set_attributes(
         # Use the latest version if version is None.
         self._version = current_version if version is None else str(version)
         self._database = current_database if database is None else str(database)
-        self._url = (
+        self._url, self._md5 = (
```

Now seeing the code and thinking a bit more: I fear it's a bit much to ask users to compute hash and add them to the yaml file. Could we make the md5 optional in the yaml? In cases users don't add hash, we'll not verify, and we can print a warning right after the download saying the integrity of the file isn't checked.

---

Another potential issue might be if the database updated the file, then the hash will no longer match. In that sense it will also be a bit annoying for us to maintain the updated md5. This is unlikely to occur with established databases like ensembl, but for instance the CellMarker database has been keep updating the file with the same url.

---

Ideally if there's a way to check if the download is interrupted, that would be the best. 🤔

---

Happy to make the md5 optional.

RE databases like the CellMarker database. IMO this is an issue on their hand. They should provide versionized URLs. We should complain about this. We could also make the md5 check optional for the CellMarker ontology, but that's a bandaid. Our implementation could still silently fail if they upgrade their ontology file

---

The problem is if the check fails, users will end up downloading a new file every single time they call `.df`, that wouldn't be great.